### PR TITLE
osd: reduce buffer pinning from EC entries

### PR DESCRIPTION
**Pull request by liewegas**
This reduces memory consumption by bufferlists (buffer_data) by about 3x.

### CODE REVIEW

File: src/osd/PGLog.h [hunk 1]
```diff
@@ -440,6 +440,9 @@ struct PGLog : DoutPrefixProvider {
 	assert(get_can_rollback_to() == head);
       }
 
+      // make sure our buffers don't pin bigger buffers
+      e.mod_desc.trim_bl();
```
Comment (jdurgin) [inline src/osd/PGLog.h]: ah, this used to be here but was accidentally removed in 5e0ec06376f832d32a6b1af390f925a59b03798f

File: src/common/config_opts.h [hunk 2]
```diff
@@ -383,6 +383,7 @@ OPTION(mon_debug_dump_transactions, OPT_BOOL, false)
 OPTION(mon_debug_dump_json, OPT_BOOL, false)
 OPTION(mon_debug_dump_location, OPT_STR, "/var/log/ceph/$cluster-$name.tdump")
 OPTION(mon_debug_no_require_luminous, OPT_BOOL, false)
+OPTION(mon_debug_no_require_bluestore_for_ec_overwrites, OPT_BOOL, false)
```
Comment (jdurgin) [inline src/common/config_opts.h]: I'd rather not add another footgun if we don't need it

File: src/osd/ECTransaction.cc [hunk 3]
```diff
@@ -166,10 +166,10 @@ void ECTransaction::generate_transactions(
       if (entry &&
 	  entry->is_modify() &&
 	  op.updated_snaps) {
-	vector<snapid_t> snaps(
-	  op.updated_snaps->second.begin(),
-	  op.updated_snaps->second.end());
-	::encode(snaps, entry->snaps);
+	bufferlist bl(op.updated_snaps->second.size() * 8 + 8);
+	::encode(op.updated_snaps->second, bl);
+	ldpp_dout(dpp, 0) << "snap bl is " << bl << dendl;
```
Comment (jdurgin) [inline src/osd/ECTransaction.cc]: dout(0) accidentally left in

**Comment by liewegas** [comment]
Even if it is marked developer only?  I needed it to narrow down whether 
this was an OSD or bluestore issue.

**Pull request merged**